.ci/oci-disconnected.sh

```diff
@@ -0,0 +1,215 @@
+
+# Define Disconnected tests environment
+export INTERNAL_REGISTRY_URL=${INTERNAL_REGISTRY_URL-"UNDEFINED"}
+export INTERNAL_REG_USERNAME=${INTERNAL_REG_USERNAME-"UNDEFINED"}
+export INTERNAL_REG_PASS="${INTERNAL_REG_PASS-"UNDEFINED"}"
+export SLACK_TOKEN="${SLACK_TOKEN-"UNDEFINED"}"
+export WORKSPACE="${WORKSPACE-"UNDEFINED"}"
+export REG_CREDS=${XDG_RUNTIME_DIR}/containers/auth.json
+export ORGANIZATION="eclipse"
+export TAG_NIGHTLY="nightly"
+
+#Stop execution on any error
+trap "catchDisconnectedJenkinsFinish" EXIT SIGINT
+
+# Catch an error after existing from jenkins Workspace
+function catchDisconnectedJenkinsFinish() {
+    EXIT_CODE=$?
+
+    if [ "$EXIT_CODE" != "0" ]; then
+      export JOB_RESULT=":alert-siren: Failed :alert-siren:"
+    else
+      export JOB_RESULT=":tada: Success :tada:"
+    fi
+
+    mkdir -p ${WORKSPACE}/artifacts
+    chectl server:logs --directory=${WORKSPACE}/artifacts
+
+    echo "[INFO] Please check Jenkins Artifacts-> ${BUILD_URL}"
+    /bin/bash "${OPERATOR_REPO}"/.github/bin/slack.sh
+
+    exit $EXIT_CODE
+}
+
+# Check if all necessary environment for disconnected test are defined
+if [[ "$WORKSPACE" == "UNDEFINED" ]]; then
+    echo "[ERROR] Jenkins Workspace env is not defined."
+    exit 1
+fi
+
+if [[ "$SLACK_TOKEN" == "UNDEFINED" ]]; then
+    echo "[ERROR] Internal registry credentials environment is not defined."
+    exit 1
+fi
+
+if [[ "$REG_CREDS" == "UNDEFINED" ]]; then
+    echo "[ERROR] Internal registry credentials environment is not defined."
+    exit 1
+fi
+
+if [[ "$INTERNAL_REGISTRY_URL" == "UNDEFINED" ]]; then
+    echo "[ERROR] Internal registry url environment is not defined."
+    exit 1
+fi
+
+if [[ "$INTERNAL_REG_USERNAME" == "UNDEFINED" ]]; then
+    echo "[ERROR] Internal registry username environment is not defined."
+    exit 1
+fi
+
+if [[ "$INTERNAL_REG_PASS" == "UNDEFINED" ]]; then
+    echo "[ERROR] Internal registry password environment is not defined."
+    exit 1
+fi
+
+# Login to internal registry using podman
+podman login -u "${INTERNAL_REG_USERNAME}" -p "${INTERNAL_REG_PASS}" --tls-verify=false ${INTERNAL_REGISTRY_URL} --authfile=${REG_CREDS}
+
+# Build che-machine-exec,che-plugin-registry and che-devfile-registry from Github Sources
+# Che-Devfile-Registry Build
+git clone git@github.com:eclipse/che-devfile-registry.git
+cd che-devfile-registry
+
+./build.sh --organization "${ORGANIZATION}" \
+           --registry "${INTERNAL_REGISTRY_URL}" \
+           --tag "${TAG_NIGHTLY}" \
+           --offline
+cd .. && rm -rf che-devfile-registry
+
+# Che-Plugin-Registry-Build
+git clone git@github.com:eclipse-che/che-plugin-registry.git
+cd che-plugin-registry
+
+export SKIP_TEST=true && ./build.sh --organization "${ORGANIZATION}" \
+           --registry "${INTERNAL_REGISTRY_URL}" \
+           --tag "${TAG_NIGHTLY}" \
+           --offline \
+           --skip-digest-generation
+
+cd .. && rm -rf che-plugin-registry
+
+# Build Che-Machine-Exec
```

Comment:
Do we need to build machine exec image if  can use nightly one.
https://quay.io/repository/eclipse/che-machine-exec?tag=latest&tab=tags

Reply:
No, my mistake. I removed it